src/lsst/cm/tools/core/panda_utils.py

```diff
@@ -128,89 +110,66 @@ def get_errors_from_jeditaskid(dbi: DbInterface, jeditaskid: int):  # pragma: no
         A list of dictionaries containing everything
         we want to update the error instance db with
     """
-    conn_status, task_status = Client.getJediTaskDetails({"jediTaskID": jeditaskid}, True, True)
+    ret = conn.get_contents_output_ext(request_id=panda_reqid, workload_id=jeditaskid)
     print(f"Checking {jeditaskid}")
+    conn_status = ret[0]
+    wmskey = list(ret[1][1].keys())[0]
+    tasks = ret[1][1][wmskey]
 
-    # grab all the PanDA IDs
-    if conn_status == 0:
-        job_ids = list(task_status["PandaID"])
-        jobs_list = []
-        if len(job_ids) > 1:
-            chunksize = 2000  # max number of allowed connections to PanDA
-            chunks = [job_ids[i : i + chunksize] for i in range(0, len(job_ids), chunksize)]
-            for chunk in chunks:
-                conn_status, ret_jobs = Client.getFullJobStatus(ids=chunk, verbose=False)
-                if conn_status == 0:
-                    jobs_list.extend(ret_jobs)
-        elif len(job_ids) == 1:
-            conn_status, ret_jobs = Client.getFullJobStatus(ids=job_ids, verbose=False)
-            if conn_status == 0:
-                jobs_list = ret_jobs
-        else:
-            return []
-            # TODO: properly address this break condition,
-            # because something went wrong
-    else:
+    if conn_status != 0:
         raise ValueError(f"Connection to Panda Failed with status {conn_status}")
 
-    # now we need to parse all the error codes for failed PandaIDs
     error_dicts = []
 
-    failed_jobs = [job for job in jobs_list if job.jobStatus == "failed"]
+    failed_jobs = [job for job in tasks if job["trans_exit_code"] != 0]
```

Comment:
For some jobs whose parent fails, the job will not be scheduled to run. Then the 'trans_exit_code' will be None. The job's status is 'Missing'.

Reply:
failed_jobs = [job for job in tasks if job["trans_exit_code"] != 0 or job["trans_exit_code"] is None] ?